src/components/PasswordInput/__test__/PasswordInput.spec.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import { render, fireEvent, screen } from "@testing-library/react";
+import { PasswordInput } from "..";
+
+jest.mock('@zxcvbn-ts/core', () => ({
+    ...jest.requireActual('@zxcvbn-ts/core'),
+    zxcvbn: jest.fn(() => ({
+        score: 0,
+        feedback: {
+            warning: '',
+            suggestions: [],
+        },
+    })),
+}));
+
+describe("PasswordInput component", () => {
+    it("renders with default props", () => {
+        render(
+            <PasswordInput
+                hidePasswordMeter
+                hint="Password must be at least 8 characters"
+                label="Password"
+                isFullWidth={true} />
+        );
+        const input = screen.getByLabelText('Password');
+        expect(input).toBeInTheDocument();
+        expect(input).toHaveAttribute('type', 'password');
+        expect(screen.getByText('Password must be at least 8 characters')).toBeInTheDocument();
+    });
+
+    it("handles onChange event", () => {
+        const handleChange = jest.fn();
+        render(
+            <PasswordInput label="Password" onChange={handleChange} />
+        );
+        const input = screen.getByLabelText("Password");
+        fireEvent.change(input, { target: { value: "password123" } });
+        expect(handleChange).toHaveBeenCalledTimes(1);
+    });
+
+    it("handles onBlur event", () => {
+        const handleBlur = jest.fn();
+        render(
+            <PasswordInput label="Password" onBlur={handleBlur} />
+        );
+        const input = screen.getByLabelText("Password");
+        fireEvent.blur(input);
+        expect(handleBlur).toHaveBeenCalledTimes(1);
+    });
```

Comment:
```suggestion
    it("handles onBlur event", async () => {
        const handleBlur = jest.fn();
        render(
          <PasswordInput label="Password" onBlur={handleBlur} />
        );
        const input = screen.getByLabelText("Password");
        await userEvent.click(input);
        await userEvent.tab();
        expect(handleBlur).toHaveBeenCalledTimes(1);
      });
```
I think this is more like a real user interaction.

instead of tab you can also do this:
`await userEvent.click(input?.parentElement || document.body);`

what do you think?
cc: @markwylde-deriv

Reply:
because users either use tab or click on outside of the input and once that happens the `onBlur` event listener will get triggered

Reply:
Yeah I agree, it's more "user like"